### PR TITLE
Release build step.

### PR DESCRIPTION
**Pull request by carlhampuswall**
Needs to be merged to verify automatic functionality, should add firmware.bin file automatically to latest created release.

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -156,6 +156,25 @@ build_flags =
 	${env:seedlabs_devkit.build_flags}
 	-D SK_NETWORKING=0
 
+[env:seedlabs_devkit_github_action_release]
+build_flags = 
+	${env:seedlabs_devkit.build_flags}
+    -D SK_ELEGANTOTA_PRO=1
+lib_deps = 
+    git@github.com:SeedLabs-it/skdk-ota-pro.git
```
Comment (brushknight) [inline platformio.ini:164]: Check for indentation

**Review by brushknight — COMMENTED** [review]
This looks great! Maybe we can also inject build datetime/commit/version?

**Comment by carlhampuswall** [comment]
> This looks great! Maybe we can also inject build datetime/commit/version?

Automatically in the release description?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -2,12 +2,60 @@ name: Release
...
+      with:
+        ssh-private-key: ${{ secrets.SKDK_OTA_PRO_PRIVKEY }}
+    
+    - name: Set env vars
+      run: |
+        echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+        echo "BUILD_TIMESTAMP=$(date -u +'%s')" >> $GITHUB_ENV
+        echo "COMMIT_HASH=$(git rev-parse HEAD)" >> $GITHUB_ENV
+
+    - name: Cache pip
```
Comment (brushknight) [inline .github/workflows/release.yml:27]: Maybe we should wrap all of that into a dockerfile?

**Review by brushknight — APPROVED** [review]
Overall looks good, if you have time, let's wrap most of the steps into a docker file that we can pre-build and cache for later in ghcr

File: .github/workflows/release.yml (hunk 2 above)
Comment (carlhampuswall) [reply .github/workflows/release.yml:27]: Will look into it!

**Review by brushknight — APPROVED** [review]

**Pull request merged**